### PR TITLE
Move config code to a separate file

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
First bit of #304. Purely moves the code, with no modifications.

### CODE REVIEW

File: trillian/ctfe/config_test.go [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ctfe
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/gogo/protobuf/proto"
```
Comment (daviddrysdale) [inline trillian/ctfe/config_test.go]: Wrong import.

Comment (Martin2112) [reply trillian/ctfe/config_test.go]: Well spotted.

Comment (pavelkalinnikov) [reply trillian/ctfe/config_test.go]: Fixed.

File: trillian/ctfe/config.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (daviddrysdale) [reply trillian/ctfe/config.go]: I'd argue keeping the existing date as it's just a move.

[the remark this answers is not in the document]
Comment (pavelkalinnikov) [reply trillian/ctfe/config.go]: Done.

File: trillian/ctfe/config_test.go [hunk 3]
```diff
@@ -0,0 +1,403 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (pavelkalinnikov) [reply trillian/ctfe/config_test.go]: Done.

File: trillian/ctfe/config.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (pavelkalinnikov) [reply trillian/ctfe/config.go]: Yes, this was originally why I left it 2016. Reverted back to 2016 again.

**Pull request merged**